scip.proto

```diff
@@ -474,6 +474,11 @@ enum SymbolRole {
   Generated = 0x10;
   // Is the symbol in test code?
   Test = 0x20;
+  // Is this a forward declaration for the symbol?
+  //
+  // Applies to languages like C, C++ and Objective-C
+  // Forward declarations should not use the Definition role.
+  ForwardDeclaration = 0x40;
```

Comment:
Given we have a `Definition` role, what do you think about calling this a `ForwardDefinition`? This would avoid the need to add separate kinds of roles for different languages like OCaml.

Reply:
Done.